src/builtins.rs

```diff
@@ -44,6 +45,16 @@ pub fn process_precompile(addr: &Address, input: &[u8]) -> MessageCallResult {
     f.execute(input, &mut output)
         .expect("No errors in precompiles");
 
+    // hijack ECDSA calls
+    if addr.to_low_u64_be() == 1 {
```

Comment:
I'm guessing this checks to make sure the address produced from ecrecover matches the address associated with the account, but really I'm confused by lines 51-54

Reply:
We want to allow Near addresses to register ECDSA keys. so we need a way to make signatures under those keys appear as if they were a different account.

So line 51 looks for an alias using the `ecdsa_map_get` function. if it finds an alias (`if let Some(addr)`), it clears the current return value, and replaces it with the target of the alias